### PR TITLE
Zy/201 bug visual images remains after reset

### LINKED ISSUE #201: BUG: visual images remains after reset
Probably need to delete the contents of the TEMP folder.

### PR DESCRIPTION
**Pull request by zheyxu**
# Pull Request Details

## What issue does this PR address

- bug visual images remains after reset

- Link to associated issue: #201

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [x] Pre-exisiting lint errors noted: [HERE]
- [x] Tested on at least one device
  - [ ] Android Phone
  - [ ] Android Emulator
  - [ ] Chrome on Android
  - [ ] Chrome
  - [ ] iOS
  - [ ] Linux
  - [x] MacOS
  - [ ] Windows
- [x] Added a reviewer

## Finalising

Once PR discussion is complete and reviewer has approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev (gjwgit)

### CODE REVIEW

File: integration_test/utils/navigator_to_page.dart
Comment (gjwgit) [inline integration_test/utils/navigator_to_page.dart:1]: Wrong file name.

File: integration_test/utils/navigator_to_page.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/// Navigate to a page in the app.
+//
+// Time-stamp: <Tuesday 2024-09-03 09:03:00 +1000 Graham Williams>
+//
+/// Copyright (C) 2023-2024, Togaware Pty Ltd
```
Comment (gjwgit) [inline integration_test/utils/navigator_to_page.dart]: Years should be 2024, not 2023-2024 - this file did not exist in 2023.

File: integration_test/dataset_demo_reset_image_test.dart [hunk 2]
```diff
@@ -0,0 +1,213 @@
+/// Test visual image disappear after the DATASET RESET.
+//
+// Time-stamp: <Tuesday 2024-09-03 10:46:15 +1000 Graham Williams>
+//
+/// Copyright (C) 2023-2024, Togaware Pty Ltd
```
Comment (gjwgit) [inline integration_test/dataset_demo_reset_image_test.dart]: Copyright date should be 2024.

File: lib/features/tree/display.dart [hunk 3]
```diff
@@ -105,12 +105,14 @@ class TreeDisplayState extends ConsumerState<TreeDisplay> {
         ? image = '$tempDir/model_tree_rpart.svg'
         : image = '$tempDir/model_tree_ctree.svg';
 
-    pages.add(
-      ImagePage(
-        title: 'TREE',
-        path: image,
-      ),
-    );
+    if (content.isNotEmpty) {
```
Comment (gjwgit) [inline lib/features/tree/display.dart]: How does this make sense? We have a general problem across the app with managing the image files and how to know they have been properly updated. I don't think this solves the issue? Explain how this fixes that issue?

**Review by gjwgit — CHANGES_REQUESTED** [review]

File: integration_test/utils/navigator_to_page.dart
Comment (zheyxu) [reply integration_test/utils/navigator_to_page.dart:1]: Thanks! But I don't know the correct name. Can you change it @gjwgit?

File: lib/widgets/image_page.dart [hunk 4]
```diff
@@ -91,21 +111,8 @@ class ImagePage extends StatelessWidget {
                 crossAxisAlignment: CrossAxisAlignment.start,
                 children: [
                   Row(
-                    // 20240726 gjw Ensure the Save button is aligned at the
```
Comment (gjwgit) [inline lib/widgets/image_page.dart:94]: Keep this comment.

**Review by gjwgit — CHANGES_REQUESTED** [review]
I am getting some odd behaviour.

Do the tests work for you?

**Comment by zheyxu** [comment]
Works for me.

**Comment by gjwgit** [comment]
When I run `make dataset_demo_reset_image.qtest` I see the following error repeated over and over:

```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following StateError was thrown running a test (but after the test had completed):
Bad state: Tried to read a provider from a ProviderContainer that was already disposed

When the exception was thrown, this was the stack:
#0      ProviderContainer.readProviderElement (package:riverpod/src/framework/container.dart:423:7)
#1      ProviderElementProxy.read (package:riverpod/src/framework/proxy_provider_listenable.dart:129:26)
#2      ProviderContainer.read (package:riverpod/src/framework/container.dart:241:21)
#3      ProviderElementBase.read (package:riverpod/src/framework/element.dart:675:23)
#4      ptyProvider.<anonymous closure>.<anonymous closure> (package:rattle/providers/pty.dart:54:9)
#15     _StringAdapterSink.add (dart:convert/string_conversion.dart:228:11)
#16     _StringAdapterSink.addSlice (dart:convert/string_conversion.dart:233:7)
#17     _Utf8ConversionSink.addSlice (dart:convert/string_conversion.dart:307:20)
#18     _Utf8ConversionSink.add (dart:convert/string_conversion.dart:300:5)
#19     _ConverterStreamEventSink.add (dart:convert/chunked_conversion.dart:69:18)
#27     CastStreamSubscription._onData (dart:_internal/async_cast.dart:85:11)
#39     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
(elided 28 frames from dart:async and package:stack_trace)
════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by junhaow1** [comment]
Hi @gjwgit  , flutter test integration_test/dataset_demo_reset_image_test.dart is working on my machine 
image

**Comment by junhaow1** [comment]
make dataset_demo_reset_image.qtest is working as well 
image

File: integration_test/utils/navigator_to_page.dart
Comment (gjwgit) [reply integration_test/utils/navigator_to_page.dart:1]: The function name is navigateToPage() So the file name is navigate_to_page.dart? That should be clear?

**Review by gjwgit — APPROVED** [review]

**Review by gjwgit — APPROVED** [review]

**Pull request merged**